router-tests/prometheus_test.go

```diff
@@ -2998,6 +2952,120 @@ func TestPrometheus(t *testing.T) {
 		})
 	})
 
+	t.Run("Collect and export OTEL metrics to Prometheus with exclude scope info", func(t *testing.T) {
+		t.Parallel()
+
+		var (
+			err        error
+			mfFiltered []*io_prometheus_client.MetricFamily
+		)
+
+		metricReaderFiltered := metric.NewManualReader()
+		promRegistryFiltered := prometheus.NewRegistry()
+
+		testenv.Run(t, &testenv.Config{
+			MetricReader:       metricReaderFiltered,
+			PrometheusRegistry: promRegistryFiltered,
+			MetricOptions: testenv.MetricOptions{
+				MetricExclusions: testenv.MetricExclusions{
+					ExcludeScopeInfo: true,
+				},
+			},
+		}, func(t *testing.T, xEnv *testenv.Environment) {
+			res := xEnv.MakeGraphQLRequestOK(testenv.GraphQLRequest{
+				Query: `query myQuery { employees { id } }`,
+			})
+
+			require.JSONEq(t, employeesIDData, res.Body)
+
+			mfFiltered, err = promRegistryFiltered.Gather()
+			require.NoError(t, err)
+
+			requestsInFlightFiltered := findMetricFamilyByName(mfFiltered, "router_http_requests_in_flight")
+			requestsInFlightMetricsFiltered := requestsInFlightFiltered.GetMetric()
+
+			require.Len(t, requestsInFlightMetricsFiltered, 2)
+
+			require.Equal(t, []*io_prometheus_client.LabelPair{
```

Comment:
Wouldn't it be easier to read if we would only test if those labels aren't included?

Reply:
I'm not sure what you mean, this is how every single test in this entire package functions. I could test for presence/absence of the otel labels by looping but it where the entire object is not too complex I don't see a big issue in comparing the full metrics captured